pkg/util/hostname.go

```diff
@@ -179,7 +179,7 @@ func GetHostnameData(ctx context.Context) (HostnameData, error) {
 	configHostnameFilepath := config.Datadog.GetString("hostname_file")
 	if configHostnameFilepath != "" {
 		log.Debug("GetHostname trying `hostname_file` config option...")
-		if fileHostnameProvider, found := hostname.ProviderCatalog["file"]; found {
```

Comment:
Does it make sense to move this file to pkg/util/hostname?

Reply:
I would very much so. I started looking into it and the change was massive. I'll open another PR dedicated to this.

Reply:
It's here now; https://github.com/DataDog/datadog-agent/pull/9738/files#diff-b2d3d13fbbcae18918c7d8e4cb45f8b467b3a8b7e78294e0b05827ffda4fb6a6R38